app/Compiler/Avro/AvroUnion.php

```diff
@@ -54,6 +54,15 @@ public function getImports(): array
 
     public function decode($data, ?string $namespace = '')
     {
+        //TODO: Handle multiple enum types
+
+        // return the first non-null type
+        foreach($this->types as $type) {
```

Comment:
Just checking my understanding:  The assumption that was breaking here was that union types were all primitives?  And this fixes it by finding the first non-null type in the union and explicitly decoding it?

Reply:
Yep that's it. I realize now why Avro requires that you send union types along with unions when representing data as JSON:
```
{
  "userId": {
    "int": 4
  }
}
```